packages/compiler/lib/decorators.ts

```diff
@@ -590,3 +594,39 @@ export function validateDecoratorParamCount(
     return;
   }
 }
+
+// -- @friendlyName decorator ---------------------
+
+const friendlyNamesKey = Symbol();
+
+export function $friendlyName(
```

Comment:
should we let that be used on every type or just model or templatable types for now?

feel like this might also be a bit releated to the `x-ms-client-name` for properties in autorest

Reply:
Forgot to add a type check, added one now.  We could definitely add support for model properties in a future PR if we decide it's worth doing.  I guess the question would be whether a `friendlyName` on a property should mean that the actual property name gets change or the `x-ms-client-name` gets used instead.